cpp/src/svm/svc_impl.h

```diff
@@ -205,16 +209,23 @@ void svcPredict(const cumlHandle &handle, math_t *input, int n_rows, int n_cols,
       cublas_handle, CUBLAS_OP_N, n_batch, model.n_support, &one, K.data(),
       n_batch, model.dual_coefs, 1, &null, y.data() + i, 1, stream));
   }
-  // Look up the label based on the value of the decision function:
-  // f(x) = sign(y(x) + b)
   math_t *labels = model.unique_labels;
   math_t b = model.b;
-  MLCommon::LinAlg::unaryOp(
-    preds, y.data(), n_rows,
-    [labels, b] __device__(math_t y) {
-      return y + b < 0 ? labels[0] : labels[1];
-    },
-    stream);
+  if (predict_class) {
+    // Look up the label based on the value of the decision function:
+    // f(x) = sign(y(x) + b)
+    MLCommon::LinAlg::unaryOp(
+      preds, y.data(), n_rows,
+      [labels, b] __device__(math_t y) {
+        return y + b < 0 ? labels[0] : labels[1];
+      },
+      stream);
+  } else {
+    // Calculate the value of the decision function: f(x) = y(x) + b
+    MLCommon::LinAlg::unaryOp(
+      preds, y.data(), n_rows,
+      [labels, b] __device__(math_t y) { return y + b; }, stream);
```

Comment:
knitpick: we don't need `labels` to be part of the device lambda's closure here, no?